composables/transaction/utils.ts

```diff
@@ -32,3 +48,36 @@ export const verifyRoyalty = (
     normalizedRoyalty,
   }
 }
+
+export function isActionValid(action: Actions): boolean {
+  const hasContent = <T>(v: T | T[]): boolean =>
+    Array.isArray(v) ? v.length > 0 : Boolean(v)
+
+  const validityMap: Record<string, (action) => boolean> = {
+    [Interaction.BUY]: (action: ActionBuy) => hasContent(action.nfts),
+    [Interaction.LIST]: (action: ActionList) => hasContent(action.token),
+    [Interaction.MINTNFT]: (action: ActionMintToken) =>
+      hasContent(action.token),
+    [Interaction.SEND]: (action: ActionSend) => Boolean(action.nftId),
+    [Interaction.CONSUME]: (action: ActionConsume) => Boolean(action.nftId),
+    [ShoppingActions.MAKE_OFFER]: (action: ActionOffer) =>
+      Boolean(action.tokenId),
+    [ShoppingActions.WITHDRAW_OFFER]: (action: ActionWithdrawOffer) =>
+      Boolean(action.nftId),
+    [Interaction.MINT]: (action: ActionMintCollection) =>
+      Boolean(action.collection),
+    [Collections.DELETE]: (action: ActionDeleteCollection) =>
+      Boolean(action.collectionId),
+    [NFTs.BURN_MULTIPLE]: (action: ActionBurnMultipleNFTs) =>
+      hasContent(action.nftIds),
+  }
+
+  const checker = validityMap[action.interaction]
+
+  if (!checker) {
+    // Assume the action is valid if there's no specific checker
+    return true
+  }
+
+  return checker(action)
```

Comment:
this ^, why do you need somehow to make verification in first place?

Reply:
if you mean why do i need to verify the content of the action to begin with, then that because the fee finding logic is getting called with incomplete actions
maybe this commnet will explain https://github.com/kodadot/nft-gallery/pull/8156#discussion_r1399389974

if you mean why do i check that checker exists before using it, that's ebcause it is possible we introduce a new interaction that isn't listed in the validityMap
in that case it makes more sense to let it through then blocking it

Reply:
> that's because it is possible we introduce a new interaction that isn't listed in the validityMap in that case it makes more sense to let it through then blocking it

this looks like a specific case, why don't we throw error then? (interaction not found?) why don't we check validity inside each interaction call like a Controller?

I mean I have nothing against if it's working as excepted

Reply:
>this looks like a specific case, why don't we throw error then?

sure. that make sense

>why don't we check validity inside each interaction call like a Controller?

do you mean move it here and early return if it's invalid?

```typescript
  const transaction = async (item: Actions, prefix = '') => {
    let api = await apiInstance.value

    if (prefix) {
      api = await apiInstanceByPrefix(prefix)
    }

    return executeAction({ item, executeTransaction, api, isLoading, status })
  }

  return {
    isLoading,
    status,
    transaction,
    blockNumber,
    isError,
  }
}
```

Reply:
Oh for now you only check tx on auto teleport, what about others? do they need to have a check beforhand?
I mean it would be better to check params, before calling `executeAction`

Reply:
only auto teleport is a "problem-maker" as far as i know

still, it does make sense to have this final verification centralized in `transaction`

Reply:
implemented, still works
re-review plz